x-pack/plugins/ml/server/models/data_recognizer/data_recognizer.js

```diff
@@ -150,6 +150,20 @@ export class DataRecognizer {
     return (resp.hits.total !== 0);
   }
 
+  async listModules() {
+    const manifestFiles = await this.loadManifestFiles();
+    const ids = manifestFiles
+      .map(({ json }) => json.id)
+      .sort((a, b) => a.localeCompare(b)); // sort as json files are read from disk and could be in any order.
+
+    const modules = [];
+    for (let i = 0; i < ids.length; i++) {
+      const module = await this.getModule(ids[i]);
```

Comment:
Nit: Using await in this `for` loop will just run each of these sequentially - which is fine - but an alternative could be to use `Promise.all` to capitalize on using `async/await` and run the tasks in parallel until they all return.

Reply:
Good spot, yes, I intentionally used for loop here to reduce the strain on kibana. each call to `getModule` files off async IO reads to load all of the related json files. All done in `Promise.all` loops.
Running those all at once for every module would cause a big spike in file reads and so might cause strain on the server.
`getModule` runs pretty quickly and has no network interaction, so I decided it should be fast enough run synchronously.